src/main/java/uk/gov/pay/api/model/CreatePaymentRequest.java

```diff
@@ -8,34 +8,65 @@
 @ApiModel(value = "CreatePaymentRequest", description = "The Payment Request Payload")
 public class CreatePaymentRequest {
 
-    public static final String AMOUNT_FIELD_NAME="amount";
-    public static final String RETURN_URL_FIELD_NAME="return_url";
-    public static final String REFERENCE_FIELD_NAME="reference";
-    public static final String DESCRIPTION_FIELD_NAME="description";
-    public static final String AGREEMENT_ID_FIELD_NAME="agreement_id";
+    public static final String AMOUNT_FIELD_NAME = "amount";
+    public static final String RETURN_URL_FIELD_NAME = "return_url";
+    public static final String REFERENCE_FIELD_NAME = "reference";
+    public static final String DESCRIPTION_FIELD_NAME = "description";
+    public static final String AGREEMENT_ID_FIELD_NAME = "agreement_id";
 
     private int amount;
     private String returnUrl;
     private String reference;
     private String description;
-    private String agreementId = null;
+    private String agreementId;
 
-    public CreatePaymentRequest(int amount, String returnUrl, String reference, String description) {
-        this(amount, returnUrl, reference, description, null);
+    public static class CreatePaymentRequestBuilder {
+        private int amount;
+        private String returnUrl;
+        private String reference;
+        private String description;
+        private String agreementId;
+
+        public CreatePaymentRequestBuilder amount(int amount) {
+            this.amount = amount;
+            return this;
+        }
+
+        public CreatePaymentRequestBuilder returnUrl(String returnUrl) {
+            this.returnUrl = returnUrl;
+            return this;
+        }
+
+        public CreatePaymentRequestBuilder reference(String reference) {
+            this.reference = reference;
+            return this;
+        }
+
+        public CreatePaymentRequestBuilder description(String description) {
+            this.description = description;
+            return this;
+        }
+
+        public CreatePaymentRequestBuilder agreementId(String agreementId) {
+            this.agreementId = agreementId;
+            return this;
+        }
+
+        public CreatePaymentRequest build() {
+            return new CreatePaymentRequest(amount, returnUrl, reference, description, agreementId);
+        }
     }
 
-    public CreatePaymentRequest(int amount, String returnUrl, String reference, String description, String agreementId) {
+    public static CreatePaymentRequestBuilder builder() {
+        return new CreatePaymentRequestBuilder();
+    }
+
+    private CreatePaymentRequest(int amount, String returnUrl, String reference, String description, String agreementId) {
```

Comment:
This now private constructor could take the builder as a single argument to make it simpler?